### ISSUE TITLE
Wrong or confusing Semantic Token Types reported

### ISSUE THREAD

**Issue by 0xalecks** [typescript-language-server/typescript-language-server#632]
When trying to customize a VSCode theme using the Semantic Highlighting, I think the TS language server is reporting wrong, or inconsistent types in certain cases:

```typescript
addToken(token: Token, test: string, foo: number): Foo {
    ...
}
```
When using the VSCode "scope inspector" tool (cmd+alt+shift+i on mac) this is what it reports as the semantic token types for the different parts of the function definition:

- "Token" -> `class`
- "string" -> no semantic types found
- "number" -> no semantic types found
- "Foo" -> `type`

<img width="655" alt="image" src="https://user-images.githubusercontent.com/99347981/203654443-4d708899-8977-419a-9d56-63d648c5f035.png">
<img width="896" alt="image" src="https://user-images.githubusercontent.com/99347981/203654460-88abd43a-8ce1-43ed-8905-e52799757a0f.png">
<img width="993" alt="image" src="https://user-images.githubusercontent.com/99347981/203654495-6f8166e7-557a-4b17-b207-3fe5de252c19.png">

Is this right? Shouldn't they all report as Type? Or at the very least, the primitives should show up somehow?

**Comment by rchl** [comment]
While this server probably has the same behavior as VSCode, if you are using VSCode then you are most likely not using this server but built-in language features instead. So I would suggest that you ask that in VSCode repo. Or possibly even directly in TypeScript repo.

That said, having no semantic tokens for basic string or number primitives sounds fine to me. Semantic tokens are suppose to extend basic syntax which should already handle those fine in any decent editor.
